src/main/java/core/basesyntax/operation/ReturnHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.transaction.FruitTransaction;
+
+public class ReturnHandler implements OperationHandler {
+    private ShopServiceImpl shopServiceImpl;

```

Comment:
You are violating the Dependency Inversion Principle from SOLID. Instead of directly depending on `ShopServiceImpl`, you should depend on the abstraction `ShopService`. This will make your `ReturnHandler` class more flexible and easier to maintain.